### PR TITLE
Add changelog for v0.1.4

### PR DESCRIPTION
**Pull request by charliermarsh**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,62 @@
...
+
+### Preview features
+
+- \[`flake8-trio`\] Implement `timeout-without-await` (`TRIO001`) ([#8439](https://github.com/astral-sh/ruff/pull/8439))
+- \[`numpy`\] Implement NumPy 2.0 migration rule (`NPY200`) ([#7702](https://github.com/astral-sh/ruff/pull/7702))
+- \[`pylint`\] Implement `bad-open-mode` (`W1501`) ([#8294](https://github.com/astral-sh/ruff/pull/8294))
+- \[`pylint`\] Implement `import-outside-toplevel` (`C0415`) rule ([#5180](https://github.com/astral-sh/ruff/pull/5180))
+- \[`pylint`\] Implement `useless-with-lock` (`W2101`) ([#8321](https://github.com/astral-sh/ruff/pull/8321))
+- \[`refurb`\] Implement `isinstance-type-none` (`FURB168`) ([#8308](https://github.com/astral-sh/ruff/pull/8308))
+- Autoformat confusable units in `RUF001`, `RUF002`, and `RUF003` ([#4430](https://github.com/astral-sh/ruff/pull/4430))
```
Comment (zanieb) [inline CHANGELOG.md:13]: What does autoformat mean?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,62 @@
...
+- Add newline after module docstrings in preview style ([#8283](https://github.com/astral-sh/ruff/pull/8283))
```
Comment (zanieb) [inline CHANGELOG.md:14]: Are we consistently putting formatter preview style here?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,62 @@
...
+- Split tuples in return positions by comma first ([#8280](https://github.com/astral-sh/ruff/pull/8280))
+- Avoid treating byte strings as docstrings ([#8350](https://github.com/astral-sh/ruff/pull/8350))
+- Add `--line-length` option to `format` command ([#8363](https://github.com/astral-sh/ruff/pull/8363))
+- Avoid parenthesizing unsplittable because of comments ([#8431](https://github.com/astral-sh/ruff/pull/8431))
+
+### Bug fixes
+
+- Respect `--force-exclude` in `lint.exclude` and `format.exclude` ([#8393](https://github.com/astral-sh/ruff/pull/8393))
+- Respect `--extend-per-file-ignores` on the CLI ([#8329](https://github.com/astral-sh/ruff/pull/8329))
+- Harmonize help commands' `--format` to `--output-format` with deprecation warnings ([#8203](https://github.com/astral-sh/ruff/pull/8203))
```
Comment (zanieb) [inline CHANGELOG.md:33]: I'm not sure this was a bug fix

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,62 @@
...
+- Improve B015 message ([#8295](https://github.com/astral-sh/ruff/pull/8295))
+- Use `fixedOverflowWidgets` for playground popover ([#8458](https://github.com/astral-sh/ruff/pull/8458))
+- Mark `byte_bounds` as a non-backwards-compatible NumPy 2.0 change ([#8474](https://github.com/astral-sh/ruff/pull/8474))
+
+### Internals
+
+- Add a dedicated cache directory per Ruff version ([#8333](https://github.com/astral-sh/ruff/pull/8333))
+- Allow selective caching for `--fix` and `--diff` ([#8316](https://github.com/astral-sh/ruff/pull/8316))
+- perf(parser): use memchr for lexing comments ([#8193](https://github.com/astral-sh/ruff/pull/8193))
+- perf(parser): use faster string parser methods ([#8227](https://github.com/astral-sh/ruff/pull/8227))
```
Comment (zanieb) [inline CHANGELOG.md:57]: Personal preference
```suggestion
- Improve performance of parsing comments ([#8193](https://github.com/astral-sh/ruff/pull/8193))
- Improve performance of parsing strings ([#8227](https://github.com/astral-sh/ruff/pull/8227))
```

**Review by zanieb — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (charliermarsh) [reply CHANGELOG.md:33]: What section would you suggest it go in?

File: CHANGELOG.md (hunk 2 above)
Comment (charliermarsh) [reply CHANGELOG.md:14]: In this changelog? Yes

File: CHANGELOG.md (hunk 3 above)
Comment (zanieb) [reply CHANGELOG.md:33]: A `CLI` section (which is one of our standard ones), perhaps with a clearer change entry e.g. "Add `--output-format` to `ruff rule` and `ruff linter`"

File: CHANGELOG.md (hunk 2 above)
Comment (zanieb) [reply CHANGELOG.md:14]: :D okay

Comment (charliermarsh) [reply CHANGELOG.md:14]: As in... I don't know if we're consistent about this across releases :)

Comment (zanieb) [reply CHANGELOG.md:14]: Well I think it's okay for now, we'll probably want to be more sophisticated someday

**Pull request merged**